plugin/src/main/java/io/fabric8/maven/plugin/ResourceMojo.java

```diff
@@ -281,7 +302,121 @@ private boolean isOpenShiftMode() {
 
     private KubernetesList convertToKubernetesResources(KubernetesList resources, KubernetesList openShiftResources) throws MojoExecutionException {
         KubernetesList kubernetesResources = removeOpenShiftObjects(resources);
-        return processOpenshiftTemplateIfProvided(openShiftResources, kubernetesResources);
+        KubernetesList kubernetesList = processOpenshiftTemplateIfProvided(openShiftResources, kubernetesResources);
+        if (kubernetesInitContainerChMod) {
+            return addPersistentVolumeInitContainerChmod(kubernetesList);
+        }
+        return kubernetesList;
+    }
+
+    private KubernetesList addPersistentVolumeInitContainerChmod(KubernetesList kubernetesList) {
+        List<HasMetadata> items = kubernetesList.getItems();
+        for (HasMetadata entity : items) {
+            if (entity instanceof Deployment) {
+                Deployment resource = (Deployment) entity;
+                DeploymentSpec spec = resource.getSpec();
+                if (spec != null) {
+                    addPersistentVolumeInitContainerChmod(entity, spec.getTemplate());
+                }
+            } else if (entity instanceof ReplicaSet) {
+                ReplicaSet resource = (ReplicaSet) entity;
+                ReplicaSetSpec spec = resource.getSpec();
+                if (spec != null) {
+                    addPersistentVolumeInitContainerChmod(entity, spec.getTemplate());
+                }
+            } else if (entity instanceof ReplicationController) {
+                ReplicationController resource = (ReplicationController) entity;
+                ReplicationControllerSpec spec = resource.getSpec();
+                if (spec != null) {
+                    addPersistentVolumeInitContainerChmod(entity, spec.getTemplate());
+                }
+            } else if (entity instanceof DeploymentConfig) {
+                DeploymentConfig resource = (DeploymentConfig) entity;
+                DeploymentConfigSpec spec = resource.getSpec();
+                if (spec != null) {
+                    addPersistentVolumeInitContainerChmod(entity, spec.getTemplate());
+                }
+            }
+        }
+        return kubernetesList;
+    }
+
+    private void addPersistentVolumeInitContainerChmod(HasMetadata entity, PodTemplateSpec template) {
+        if (template != null) {
+            PodSpec podSpec = template.getSpec();
+            if (podSpec != null) {
+                Map<String, String> nameToMount = new TreeMap<>();
+                List<Volume> volumes = podSpec.getVolumes();
+                if (volumes != null) {
+                    for (Volume volume : volumes) {
+                        PersistentVolumeClaimVolumeSource persistentVolumeClaim = volume.getPersistentVolumeClaim();
+                        if (persistentVolumeClaim != null) {
+                            String name = volume.getName();
+                            nameToMount.put(name, "");
```

Comment:
Why do you populate the Map with an empty path ? afais it is only used to detect whether there are PVC which should be handled. If such a volume isn't refered to in the volume mount sections below (line 364), this empty string will end up in the chmod command (since it doesnt get overwritten), causing very likely an error.